src/code/db_camera.c

```diff
@@ -1,284 +1,1453 @@
 typedef struct {
-    u8 state;
-    s16 mode;
-    CutsceneCameraPoint* eyePoints;
-    CutsceneCameraPoint* atPoints;
-    s16 frameCnt;
-    s16 csPointsCnt;
-} DBCSEditorSlot;
+    /* 0x0000 */ s16 mode;
+    /* 0x0002 */ s16 nFrames;
+    /* 0x0004 */ s16 nPoints;
+    /* 0x0006 */ s16 unkIdx;
+    /* 0x0008 */ s16 unk_08;
+    /* 0x000A */ s16 unk_0A;
+    /* 0x000C */ s32 unk_0C; // indicates position vs lookAt?
+    /* 0x0010 */ char unk_10[0x14];
+    /* 0x0024 */ CutsceneCameraPoint position[129];
+    /* 0x0834 */ CutsceneCameraPoint lookAt[129];
+    /* 0x1044 */ s16 demoCtrlMenu;
+    /* 0x1046 */ s16 demoCtrlActionIdx; // e (?), s (save), l (load), c (clear)
+    /* 0x1048 */ s16 demoCtrlToggleSwitch;
+    /* 0x104A */ s16 unk_104A;
+    /* 0x104C */ s16 unk_104C;
+    /* 0x104E */ s16 unk_104E;
+} DbCameraSub; // size = 0x1050
 
-extern DBCSEditorSlot sDbgEditorSlots[];
-extern char sDbgEditorStates[];
+typedef struct {
+    /* 0x0000 */ u32 unk_00;
+    /* 0x0004 */ Vec3f at;
+    /* 0x0010 */ Vec3f eye;
+    /* 0x001C */ Vec3f unk_1C;
+    /* 0x0028 */ char unk_28[0xC];
+    /* 0x0034 */ s32 unk_34;
+    /* 0x0038 */ s32 unk_38;
+    /* 0x003C */ s32 unk_3C;
+    /* 0x0040 */ s32 unk_40;
+    /* 0x0044 */ s32 unk_44;
+    /* 0x0048 */ f32 fov;
+    /* 0x004C */ s16 roll;
+    /* 0x004E */ char unk_4E[0x2];
+    /* 0x0050 */ f32 unk_50;
+    /* 0x0054 */ char unk_54[0x60 - 0x54];
+    /* 0x0060 */ Vec3f unk_60;
+    /* 0x006C */ Vec3f unk_6C;
+    /* 0x0078 */ s16 unk_78;
+    /* 0x007A */ s16 unk_7A;
+    /* 0x007C */ DbCameraSub sub;
+} DbCamera; // size = 0x10CC
+
+typedef struct {
+    /* 0x00 */ char letter;
+    /* 0x01 */ u8 unk_01;
+    /* 0x02 */ s16 mode;
+    /* 0x04 */ CutsceneCameraPoint* position;
+    /* 0x08 */ CutsceneCameraPoint* lookAt;
+    /* 0x0C */ s16 nFrames;
+    /* 0x0E */ s16 nPoints;
+} DbCameraCut; // size = 0x10
+
+typedef struct {
+    /* 0x00 */ f32 curFrame;
+    /* 0x04 */ f32 unk_04; // frame count?
+    /* 0x08 */ s16 keyframe;
+    /* 0x0A */ s16 unk_0A;
+    /* 0x0C */ s16 unk_0C;
+    /* 0x10 */ Vec3f positionPos; // confusing name
+    /* 0x1C */ Vec3f lookAtPos;
+    /* 0x28 */ f32 roll;
+    /* 0x2C */ f32 fov;
+} DbCameraAnim; // size = 0x30
+
+char* D_8012CEE0[] = { "\x8Cｷ-ﾌﾚ-ﾑ\x8Dｶﾞ" };
+const char* D_8012CEE4 = "\x8Dﾀﾘﾏｾﾝ｡";
+const char* D_8012CEE8 = "\x8Dｻｲｾｲﾃﾞｷﾏｾﾝ";
+const char* D_8012CEEC = "\x8Dｻｲｾｲｼｭｳﾘｮｳ";
+const char* D_8012CEF0 = "\x8Dｻｲｾｲﾁｭｳ!";
+
+const char* D_8012CEF4 = "DEMO CAMERA TOOL";
+const char* D_8012CEF8[] = { "\x8Dﾓｳﾊｲﾘﾏｾﾝ", "\x8Dﾄｳﾛｸ   ﾃﾝﾒ", "\x8Dﾍﾝｺｳ   /  " };
+const char* D_8012CF04 = ">            >";
+const char* D_8012CF08 = "<            <";
+const char* D_8012CF0C = "<          >";
+const char* D_8012CF10 = "\x8C*ﾌﾟﾚｲﾔ-*";
+const char* D_8012CF14 = "\x8C\Eﾓ-ﾄﾞ\x8D ｿｳﾀｲ";
+const char* D_8012CF18[] = { "\x8C\Eﾓ-ﾄﾞ\x8Dｾﾞｯﾀｲ", "\x8Dｶﾞﾒﾝ\x8C   ﾃﾞﾓ", "\x8Dｶﾞﾒﾝ   ﾌﾂｳ" };
+const char* D_8012CF24[] = { "\x8DPｼﾞｶﾝ  MAX", "\x8Cﾘﾝｸ\x8D    ｷｵｸ", "\x8Cﾘﾝｸ\x8D     ﾑｼ" };
```

Comment:
most if not all of these should be able to be inlined where they are actually used, right? or is there an issue with their ordering? didnt look at it too closely

Reply:
I'm kind of confused too, all of these strings appear to be unreferenced in the code itself, unless they get used in another file..?

Reply:
Had a quick look at the map file. all of these strings appear to be stored in the .data section of db_camera, which is a fairly unusual (but not unprecedented) situation. A bunch of these strings belong to DbCamera_Update, which hasn't been attempted, which is likely why they haven't been inlined.

Reply:
As @mzxrules mentioned, these are mostly related to `DbCamera_Update` there was a discussion on Discord about turning some values into 2D arrays but I think it would be better if it was done by whoever wants to decompile `DbCamera_Update` since I'm not sure which value belongs to which 2D array.